### PR TITLE
traefik: 2.2.8 -> 2.2.11, pin to go 1.14

### PR DESCRIPTION
**Pull request by amaxine**
###### Motivation for this change
Version bump. Pinning to 1.14 based on https://github.com/containous/traefik/issues/7234 and since the x.509 changes in 1.15 would result in probably unhappy changes in behaviour for anyone running this in production.

Also fixes passing build flags so that traefik no longer reports its version as `dev`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by endocrimes** [comment]
Result of `nixpkgs-review pr 97568` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>traefik</li>
  </ul>
</details>

**Review by endocrimes — APPROVED** [review]

**Review by zowoq — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1511,7 +1511,7 @@ in
 
   caddy = callPackage ../servers/caddy { buildGoModule = buildGo114Module; };  # https://github.com/lucas-clemente/quic-go/issues/2614
   caddy1 = callPackage ../servers/caddy/v1.nix { buildGoModule = buildGo114Module; };
-  traefik = callPackage ../servers/traefik { };
+  traefik = callPackage ../servers/traefik { buildGoModule = buildGo114Module; };
```
Comment (kalbasit) [inline pkgs/top-level/all-packages.nix:1514]: Could you add a comment above buildGoModule explaining why it was pinned? This should help next maintainer to quickly know why without having to bisect.

**Review by kalbasit — APPROVED** [review]
minor comment, otherwise LGTM

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (maxeaubrey) [reply pkgs/top-level/all-packages.nix:1517]: Good point, added!

**Pull request merged**